fennel/featuresets/featureset.py

```diff
@@ -551,6 +562,70 @@ def _add_feature_names_as_attributes(self):
     def all(self) -> List[Feature]:
         return self._features
 
+    def _get_expression_extractors(self) -> List[Extractor]:
+        extractors = []
+        for feature in self._features:
+            expr = feature._expr
+            if expr is None:
+                continue
+            if not isinstance(expr, Expr):
+                raise TypeError(
+                    f"Expected an Expr object for feature {feature.name} "
+                    f"but found {type(expr)}"
+                )
+
+            ref_extractor = FetchReferences()
+            user_defined_inputs = ref_extractor.fetch(expr)
+            extractor = Extractor(
+                name=f"_fennel_expr_{feature.fqn()}",
+                extractor_type=ExtractorType.EXPR,
+                user_defined_inputs=list(user_defined_inputs),
+                user_defined_outputs=[feature],
+                version=feature.ref_version or 0,
+                func=None,
+                derived_extractor_info=None,
+                depends_on=None,
+                env=feature.ref_env,
+            )
+            # Converting the string in extractor inputs decorator to Feature object
+            inputs = []
+            for input in extractor.user_defined_inputs:
+                if isinstance(input, str):
+                    try:
+                        inputs.append(self._feature_map[input])
+                    except KeyError:
+                        raise ValueError(
+                            f"When using F(<feature_name>) for expressions, one can only choose from the features defined in the current featureset. "
```

Comment:
FYI - this error will change when we change `F` to `col`

Reply:
The language also need a bit work:

`expression refers to feature col("{}") not present in the featureset {}`